docs/cli.rst

```diff
@@ -37,30 +37,74 @@ The builtin REPL supports basic code completion suggestions, syntax highlighting
 Start an nREPL Session
 ----------------------
 
+An nREPL server provides an interactive REPL environment for remote code execution and development from an editor.
+
 Basilisp's CLI incorporates an nREPL server adapted from `nbb <https://github.com/babashka/nbb>`_.
 
-To start the server from the command line use the following command
+Start from an editor with a Clojure extension supporting Basilisp
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+.. note::
+   The extension's ``Jack-in`` command is used to start an nREPL session and connect your editor to it.
+
+With Emacs `CIDER v1.14 <https://docs.cider.mx/cider/platforms/basilisp.html>`_ and Visual Studio Code `Calva v2.0.453 <https://calva.io/basilisp/>`_ or later, you can ``Jack-in`` to a Basilisp project directly. The extensions also recognize Basilisp ``.lpy`` files as Clojure files.
+
+To ``Jack-in`` to a Basilisp project
+
+1. Ensure that a ``basilisp.edn`` file is present at the root of your project, even if it is empty.
+2. Run the ``Jack-in`` command in your editor and select ``Basilisp`` if prompted. The Editor should then start the server and connect to it.
+
+Start from an editor with a Clojure extension not yet supporting Basilisp
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+.. note::
+   The extension's ``connect`` command is used to connect your editor to a running nREPL server. It requires the port number where the server is running at.
+
+.. note::
+   Basilisp code files use the ``.lpy`` filename suffix. You might need to adjust your editor's extension settings to recognize this suffix as a Clojure file. Alternatively, you can evaluate code inside ``.clj`` files, though importing these files from other Basilisp files might not be possible due to the different file extension.
+
+If your editor extension does not yet support Basilisp, or if you prefer more control over the nREPL server, you can start the server from the command line and connect to using your extension's ``connect`` command.
+
+The nREPL server when started will provide the host and port number it is listening on.
+
+To view available command line options, use
+
+.. code-block:: bash
+
+   basilisp nrepl-server -h
```

Comment:
A couple of style nits:
 * I'm typically using a blank line between directives (e.g. `.. note::`) and their contents in these documents -- can you match that with these `..note::` directives?
 * I've been writing one sentence per line. This helps with minimizing the diff on edits.

Reply:
Updated, these guidelines add a nice feel to it :)